gloo/docs/user_guides/tls_setup.md

```diff
@@ -0,0 +1,317 @@
+---
+title: Setting up Server TLS
+weight: 50
+description: Understanding how to set up TLS for Gloo
+---
+
+## Configuring TLS for Gloo
+
+We can configure HTTPS for our Gloo services to encrypt traffic coming from external clients. We can also configure Gloo to do mTLS with external clients as well. In this document, we'll explore configuring Gloo for server TLS.
+
+## Server TLS
+Gloo supports server-side TLS where the server presents a certificate to the client based on the domain for what the client asked. This means we can support multiple virtual hosts on a single port and use SNI to determine what certificate to serve depending what domain the client is requesting. In Gloo, we associate our TLS configuration with a specific VirtualService which can then describe which SNI hosts would be candidates for both the TLS certificates as well as the routing rules that are defined in the VirtualService. Let's look at setting up TLS.
+
+### Prepare our environment
```

Comment:
I'd call this something like `setting up example environment` as looks like instructions have nothing to do with getting Gloo ready for TLS; you're just creating an example to show how to do it... Just need clearer header and some intro text to make differentiate what you're setting up for the example AND what all Gloo users would need to prep, if anything, for their services.